# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/10 16:57:46-02:00 marcelo@logos.cnet 
#   Alan Cox: sdla_xfer lack of bounds checking, reported by Coverity (from 2.6.10-ac)
# 
# drivers/net/wan/sdla.c
#   2005/01/10 16:56:45-02:00 marcelo@logos.cnet +2 -0
#   Alan Cox: sdla_xfer lack of bounds checking, reported by Coverity (from 2.6.10-ac)
#   ,.
# 
diff -Nru a/drivers/net/wan/sdla.c b/drivers/net/wan/sdla.c
--- a/drivers/net/wan/sdla.c	2005-01-13 08:41:42 -08:00
+++ b/drivers/net/wan/sdla.c	2005-01-13 08:41:42 -08:00
@@ -1300,6 +1300,8 @@
 
 		case SDLA_WRITEMEM:
 		case SDLA_READMEM:
+			if(!capable(CAP_SYS_RAWIO))
+				return -EPERM;
 			return(sdla_xfer(dev, (struct sdla_mem *)ifr->ifr_data, cmd == SDLA_READMEM));
 
 		case SDLA_START:
